### PR TITLE
[DataGridPremium] Support `valueFormatter` on the grouping column

### LINKED ISSUE #3980: [DataGridPro] Support `valueFormatter` for the grouping columns
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

I'm making grouping for a field, name of field is **parentCoverageCode**. 
image

Everything is good but TRF value is a key, I want to show as  **Traffic(3)** instead of **TRF(3)** for grouping name,  so value of **parentCoverageName** variable.

I used "groupingValueGetter", but it changed grouping field's id , it was **"auto-generated-row-parentCoverageCode/Traffic
"**, i dont want it. Maybe I can use https://mui.com/components/data-grid/group-pivot/#custom-grouping-column but it has high cost for simple process.

I didn't find any solution for this case,  if there isn't, maybe you can add new prop that naming groupingValueFormatter 
Thank you.

### Examples 🌈

_No response_

### Motivation 🔦

_No response_

### Order ID 💳 (optional)

37219

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #3980

### CODE REVIEW

File: packages/grid/x-data-grid-pro/src/internals/hooks/features/rowGrouping/createGroupingColDef.tsx [hunk 1]
```diff
@@ -198,15 +198,15 @@ export const createGroupingColDefForOneGroupingCriteria = ({
           return <GridGroupingColumnLeafCell {...leafParams} />;
         }
 
-        return null;
+        return '';
```
Comment (flaviendelangle) [inline packages/grid/x-data-grid-pro/src/internals/hooks/features/rowGrouping/createGroupingColDef.tsx:201]: When `renderCell` returns `null`, the grid fallbacks on the formatted value (I want to change that on v6).
So if a user sets a value formatter on a grouping column with `rowGroupingColumnMode="multiple"`, it renders also on the cells of the other grouping criterias.

Returning `''` here avoid this fallback and make sure that the value returned by `renderCell` is always the one displayed.

File: docs/data/data-grid/group-pivot/RowGroupingCustomGroupingColDefCallback.tsx [hunk 2]
```diff
@@ -98,13 +99,26 @@ export default function RowGroupingCustomGroupingColDefCallback() {
...
+            if (params.fields.includes('director')) {
+              override.headerName = 'Director';
+            }
+
+            override.valueFormatter = (valueFormatterParams) => {
+              const rowNode = apiRef.current.getRowNode(valueFormatterParams.id!);
+              if (rowNode?.groupingField === 'year') {
+                const value = rowNode.groupingKey;
+                if (value && typeof value === 'number') {
+                  return value.toLocaleString();
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply docs/data/data-grid/group-pivot/RowGroupingCustomGroupingColDefCallback.tsx:113]: The `<Chip />` text en effect should be modified too

[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/data-grid/group-pivot/RowGroupingCustomGroupingColDefCallback.tsx:113]: In that case I will keep the old columns and add a `by ...` on the director :

[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/data-grid/group-pivot/RowGroupingCustomGroupingColDefCallback.tsx]: Done

**Review by alexfauquette — APPROVED** [review]

**Pull request merged**